packages/dht/src/connection/WebRTC/WebRtcConnector.ts

```diff
@@ -0,0 +1,274 @@
+require('setimmediate')
+import { EventEmitter } from "events"
+import { Event as ConnectionSourceEvents, IConnectionSource } from '../IConnectionSource'
+import {
+    HandshakeMessage,
+    IceCandidate,
+    Message,
+    MessageType,
+    PeerDescriptor,
+    RtcAnswer,
+    RtcOffer, WebRtcConnectionRequest
+} from '../../proto/DhtRpc'
+import { Empty } from '../../proto/google/protobuf/empty'
+import { ITransport } from '../../transport/ITransport'
+import { RoutingRpcCommunicator } from '../../transport/RoutingRpcCommunicator'
+import { ConnectionType, Event as ConnectionEvents, IConnection } from '../IConnection'
+import { NodeWebRtcConnection } from './NodeWebRtcConnection'
+import { RemoteWebrtcConnector } from './RemoteWebrtcConnector'
+import { WebRtcConnectorClient } from '../../proto/DhtRpc.client'
+import { Event as IWebRtcEvent } from './IWebRtcConnection'
+import { PeerID } from '../../helpers/PeerID'
+import { DescriptionType } from 'node-datachannel'
+import crypto from "crypto"
+import { TODO } from '../../types'
+import { DeferredConnection } from '../DeferredConnection'
+import { Logger } from '../../helpers/Logger'
+import * as Err from '../../helpers/errors'
+import { IWebRtcConnector } from "../../proto/DhtRpc.server"
+import { ServerCallContext } from '@protobuf-ts/runtime-rpc'
+
+const logger = new Logger(module)
+
+export interface WebRtcConnectorConfig {
+    rpcTransport: ITransport,
+    fnCanConnect: (peerDescriptor: PeerDescriptor) => boolean,
+    fnGetConnection: (peerDescriptor: PeerDescriptor) => IConnection | null,
+    fnAddConnection: (peerDescriptor: PeerDescriptor, connection: IConnection) => boolean
+}
+
+export class WebRtcConnector extends EventEmitter implements IConnectionSource, IWebRtcConnector {
+    private WEBRTC_CONNECTOR_APP_ID = 'webrtc_connector'
+    private ownPeerDescriptor: PeerDescriptor | null = null
+    private rpcCommunicator: RoutingRpcCommunicator
+    private rpcTransport: ITransport
+    private getManagerConnection: (peerDescriptor: PeerDescriptor) => IConnection | null
+    private addManagerConnection: (peerDescriptor: PeerDescriptor, connection: IConnection) => boolean
+    constructor(private config: WebRtcConnectorConfig) {
```

Comment:
```suggestion
    constructor(config: WebRtcConnectorConfig) {
```

The field is never read (it is used in constructor, but no need to store it as a field of `WebRtcConnector`)